dask_expr/expr.py

```diff
@@ -635,31 +635,32 @@ def visualize(self, filename="dask-expr.svg", format=None, **kwargs):
         graphviz_to_file(g, filename, format)
         return g
 
-    def find_subtype(self, subtype: type) -> Set[Expr]:
-        """Search the expression graph for a specific `Expr` subtype
+    def find_operations(self, operation: type) -> list[Expr]:
+        """Search the expression graph for a specific operation type
 
         Parameters
         ----------
-        subtype
-            The `Expr` subtype to search for.
+        operation
+            The operation type to search for.
 
         Returns
         -------
-        nodes: set
-            Set of `subtype` instances.
+        nodes
+            List of `operation` instances. Ordering corresponds
+            to a depth-first search of the expression graph.
         """
 
-        assert issubclass(subtype, Expr), "`subtype` must be `Expr` subclass"
+        assert issubclass(operation, Expr), "`operation` must be `Expr` subclass"
         stack = [self]
-        seen, nodes = set(), set()
+        seen, nodes = set(), []
         while stack:
             node = stack.pop()
             if node._name in seen:
                 continue
             seen.add(node._name)
 
-            if isinstance(node, subtype):
-                nodes.add(node)
+            if isinstance(node, operation):
+                nodes.append(node)
```

Comment:
Just to be clear, my recommendation was ...

```suggestion
                yield node
```

I'm open to not doing this though if you'd like to avoid iterators.  I don't have a strong desire here and maybe this is too complicated.

Reply:
yes, sorry. I was adding commits to do both. Sorry, you saw only the first commit so quickly :)

Reply:
To clarify, my latest commit returns a generator, and the one you are commenting on here returns a list (to only capture depth-first ordering).  I have no strong preference either way.